In [ ]:
import pandas
import pickle
import numpy as np
import matplotlib.pyplot as plt
import sklearn
from sklearn import linear_model
#%matplotlib outline

EXPERIMENT_DATA = pickle.load(open('EXPERIMENT_SET_pandas.pkl', 'rb'))
EVALUATION_SET = pickle.load(open('EVALUATION_SET_pandas.pkl', 'rb'))
#EXPERIMENT_DATA = EXPERIMENT_DATA[EXPERIMENT_DATA["GRAD"] == "YES"]

t1 = EXPERIMENT_DATA[EXPERIMENT_DATA["REPNO"] == 1]
t2 = EXPERIMENT_DATA[EXPERIMENT_DATA["REPNO"] == 2]
t3 = EXPERIMENT_DATA[EXPERIMENT_DATA["REPNO"] == 3]
print(len(t1), len(t2), len(t3))


EXPERIMENT_DATA = EXPERIMENT_DATA[EXPERIMENT_DATA["GRAD"] == "YES"]

t1 = EXPERIMENT_DATA[EXPERIMENT_DATA["REPNO"] == 1]
t2 = EXPERIMENT_DATA[EXPERIMENT_DATA["REPNO"] == 2]
t3 = EXPERIMENT_DATA[EXPERIMENT_DATA["REPNO"] == 3]
print(len(t1), len(t2), len(t3))



def plotRegression():
	bagSold = np.asarray(EXPERIMENT_DATA["BAGSOLD"]).reshape(-1, 1).astype(np.float)
	rm = np.asarray(EXPERIMENT_DATA["RM"]).reshape(-1,1).astype(np.float)
	plt.subplot(121)
	plt.title('Linear Regression RM versus Bagsold')
	plt.plot(rm,bagSold,'ro')
	regr = linear_model.LinearRegression()
	regr.fit(rm,bagSold)
	plt.plot(rm,regr.predict(rm),color = 'red')

	plt.subplot(122)
	plt.title('Linear Regression RM versus log of Bagsold')
	bagSold = np.log(bagSold)
	plt.plot(rm,bagSold,'ro')
	regr = linear_model.LinearRegression()
	regr.fit(rm,bagSold)
	plt.plot(rm,regr.predict(rm),color = 'red')

	plt.show()


In [2]:
plotRegression()

In [4]:
location_map = set(EXPERIMENT_DATA["LOCATION"])

In [6]:
print(len(location_map))

140


### Analysis

From the preliminary analysis, we find that the number of different locateion in the dataset is 140. The location in the dataset is encoded as a 4-digit number. We first expected that we can group the quality of the species based on the location parameters. We then plot the average of __RM__ and __YIELD__ for each location, which is shown below: 

In [ ]:
location_map = list(set(EXPERIMENT_DATA["LOCATION"]))
location_map.sort()

list_location = []
list_avg_rm = []
list_avg_yield = []
for val in location_map:
    avg_rm = np.average(EXPERIMENT_DATA[EXPERIMENT_DATA["LOCATION"] == str(val)]["RM"])
    avg_yield = np.average(EXPERIMENT_DATA[EXPERIMENT_DATA["LOCATION"] == str(val)]["YIELD"])
    list_location.append(str(val))
    list_avg_rm.append(avg_rm)
    list_avg_yield.append(avg_yield)
    print("{} = {},{}".format(val,avg_rm,avg_yield))
    
plt.title("Average RM and YIELD for each location")
plt.plot(list_avg_rm, list_avg_yield, 'ro')

for i, txt in enumerate(list_location):
    if int(txt) <= 1000:
        plt.annotate(txt, (list_avg_rm[i],list_avg_yield[i]), color="blue")
    elif int(txt) <= 2000:
        plt.annotate(txt, (list_avg_rm[i],list_avg_yield[i]), color="red")
    elif int(txt) <= 3000:
        plt.annotate(txt, (list_avg_rm[i],list_avg_yield[i]), color="green")
    elif int(txt) <= 4000:
        plt.annotate(txt, (list_avg_rm[i],list_avg_yield[i]), color="black")
    elif int(txt) <= 5000:
        plt.annotate(txt, (list_avg_rm[i],list_avg_yield[i]), color="orange")
    else:
        plt.annotate(txt, (list_avg_rm[i],list_avg_yield[i]), color="purple")
plt.show()


0010 = 0.7999999999999999,41.02006987
0020 = 0.5,51.1125208
0030 = 0.5,32.824989825
0101 = 1.2,33.832471325
0103 = 1.2,55.8782157
0107 = 1.2,43.534346074999995
0110 = 0.9666666666666667,44.868794699999995
0119 = 1.2,43.772117825
0120 = 1.08,53.97745452
0130 = 0.9666666666666667,52.842995383333324
0140 = 1.2,46.978727625000005
0150 = 1.0,31.5415082
0160 = 0.5,47.7500359
0510 = 0.9666666666666667,57.99771501666666
0512 = 1.2,42.089150489999994
0513 = 1.4000000000000001,53.20812888333333
0517 = 1.2000000000000002,65.49310085
0520 = 1.35,66.3760272625
0550 = 1.1875,52.11954688749999
1102 = 1.2,46.9666457
1110 = 1.777049180327869,51.72896262459017
1120 = 1.3285714285714287,47.758214421428576
1130 = 1.0,65.48365836666666
1140 = 2.1166666666666667,62.34210493333334
1210 = 1.6285714285714286,52.84316886428571
1220 = 1.95,57.325961575
1230 = 2.0555555555555554,54.148868755555554
1240 = 2.0902777777777777,53.11554645277778
1241 = 2.1,51.57825005
1250 = 2.008163265306122,50.58025422448979
1260 = 